### PR TITLE
OpenAPI draft specification

### PR DESCRIPTION
**Pull request by francescopersico**

### CODE REVIEW

File: openapi/index.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    name: Ocp-Apim-Subscription-Key
+    in: header
+consumes:
+  - application/json
+produces:
+  - application/json
+paths:
+  "/redeemed":
+    post:
+      operationId: addRedeemBonuses
```
Comment (gunzip) [inline openapi/index.yaml:25]:
```suggestion
      operationId: setBonusesAsRedeemed
```

File: openapi/index.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        description: |-
+          Date-time the batch has been generated.
+      items:
+        type: array
+        items:
+          $ref: "#/definitions/BonusID"
+      total_redeemed_bonuses:
+        type: number
+        description: |-
+          The total number of bonuses activated (including the ones returned in the `items` property).
```
Comment (gunzip) [inline openapi/index.yaml:75]:
```suggestion
          The total number of bonuses redeemed (including the ones returned in the `items` property).
```

File: openapi/index.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
+swagger: "2.0"
+info:
+  version: 1.0.0-draft
+  title: IO BONUS API
+  description: >
+    # IO API for Bonus Vacanze
+host: api-gad.io.italia.it
+basePath: "/api/bonus/v1"
```
Comment (cloudify) [inline openapi/index.yaml:8]:
```suggestion
basePath: "/api/bonus-vacanze/v1"
```

File: openapi/index.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      operationId: setBonusesAsRedeemed
+      summary: Set bonuses status to redeemed
+      description: TDB
+      parameters:
+        - name: redeemed_bonuses
+          in: body
+          schema:
+            $ref: "#/definitions/RedeemedBonuses"
+      responses:
+        "202":
```
Comment (cloudify) [inline openapi/index.yaml:34]: no schema here?

File: openapi/index.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+          description: Request accepted
+        "400":
+          description: Invalid payload
+          schema:
+            $ref: "#/definitions/ProblemJson"
+        "401":
+          description: Unauthorized
+        "429":
```
Comment (cloudify) [inline openapi/index.yaml:42]: does the APIM replies with any specific headers to instruct the client when to retry?

File: openapi/index.yaml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      generated_at:
+        type: string
+        format: date-time
+        description: |-
+          Date-time the batch has been generated.
+      items:
+        type: array
+        items:
+          $ref: "#/definitions/BonusID"
+      total_redeemed_bonuses:
```
Comment (cloudify) [inline openapi/index.yaml:72]: I would keep this kind of check out for now, we need to understand with them first what kind of checks are possible/makes sense, let's keep it simple

File: openapi/index.yaml (hunk 5 above)
Comment (francescopersico) [reply openapi/index.yaml:42]: Not at the moment. But we can output the number of remaining calls and the retry-after period.

Comment (francescopersico) [reply openapi/index.yaml:42]: https://azure.microsoft.com/en-in/updates/azure-api-management-update-january-2020/

File: openapi/index.yaml (hunk 6 above)
Comment (gunzip) [reply openapi/index.yaml:72]: is this a cross partition query for us ? what's ou partition key for the bonus collection ?

File: openapi/index.yaml (hunk 4 above)
Comment (gunzip) [reply openapi/index.yaml:34]: 202 has no payload in our responses

Comment (francescopersico) [reply openapi/index.yaml:34]: We need to add an optional payload. Do you think it can cause problems?

Comment (gunzip) [reply openapi/index.yaml:34]: can we use 201 instead ? (which already has a payload)

**Pull request merged**